File: src/programmers/java/level3/네트워크.java
package programmers.level3;

public class 네트워크 {
    public static void main(String[] args) {
        int n = 3;
        int[][] computers = {{1, 1, 0}, {1, 1, 0}, {0, 0, 1}};
        solution(n, computers);
    }

    public static int solution(int n, int[][] computers) {
        boolean[] check = new boolean[n];
        int answer = 0;

        for (int i = 0; i < n; i++) {
            if (!check[i]) {
                dfs(computers, check, i);
                answer++;
            }
        }

        return answer;
    }

    private static void dfs(int[][] computers, boolean[] check, int num) {
        check[num] = true;

        for (int i = 0; i < computers[num].length; i++) {
            if (computers[num][i] == 1 && !check[i])
                dfs(computers, check, i);
        }
    }
}

File: src/programmers/javascript/level3/네트워크.js
// DFS
function solution(n, computers) {
    let visited = new Array(n).fill(false);
    let answer = 0;
    
    const DFS = (start) => {
        for(let i = 0; i < n; i++){
            if(computers[start][i] === 1 && !visited[i]){
                visited[i] = true;
                DFS(i);
            }
        }
    }
    
    for(let i = 0; i < n; i++){
        if(!visited[i]){
            DFS(i);
            answer++;
        }
    }
    
    return answer;
}